flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -916,11 +931,15 @@ public final void cleanUp(Throwable throwable) throws Exception {
         suppressedException =
                 runAndSuppressThrowable(this::releaseOutputResources, suppressedException);
 
-        suppressedException =
-                runAndSuppressThrowable(channelIOExecutor::shutdown, suppressedException);
-
-        suppressedException = runAndSuppressThrowable(mailboxProcessor::close, suppressedException);
+        if (channelIOExecutor != null) {
+            suppressedException =
+                    runAndSuppressThrowable(channelIOExecutor::shutdown, suppressedException);
+        }
```

Comment:
I think there are other resources that might be null as well. Not just 
`channelIOExecutor` and `mailboxProcessor`

Let's take `releaseOutputResources` for example, 
both operatorChain and recordWriter can possibly be null if thrown exception from the constructor?

Reply:
Yes, of course, there are more possible NPE but it is not problem since all of that NPE could happen only inside runAndSuppressThrowable which helps to handle it correctly. at least, all variables will checked

Reply:
er... I think `runAndSuppressThrowable` is not designed for this purpose, but coincidentally yes ^-^.